libraries/stdlib/jvm/src/kotlin/reflect/TypesJVM.kt

```diff
@@ -136,8 +136,8 @@ private class TypeVariableImpl(private val typeParameter: KTypeParameter) : Type
     @Suppress("VIRTUAL_MEMBER_HIDDEN")
     fun getDeclaredAnnotations(): Array<Annotation> = emptyArray()
 
-    @Suppress("VIRTUAL_MEMBER_HIDDEN")
-    fun getAnnotatedBounds(): Array<Annotation> = emptyArray()
+    // There is also [getAnnotatedBounds] which returns an array of [AnnotatedType]; because [AnnotatedType] is JDK 8+,
+    // we can't declare that method here, so we leave it unimplemented to throw an exception at runtime if called.
```

Comment:
The comment is unclear about why we can't declare it here, since we're compiling reflect with JDK 8.

Reply:
Added a few more words